### ISSUE TITLE
--inline-vue target lib not working

### ISSUE THREAD

**Issue by dertuerke** [vuejs/vue-cli#5204]
### Version
4.2.2

### Environment info
```
System:
    OS: Windows 10 10.0.18362
    CPU: (4) x64 Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
  Binaries:
    Node: 12.15.0 - C:\Program Files\nodejs\node.EXE     
    Yarn: Not Found
    npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD        
  Browsers:
    Edge: 44.18362.449.0
  npmPackages:
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0        
    @vue/babel-plugin-transform-vue-jsx:  1.1.2 
    @vue/babel-preset-app:  4.2.2 
    @vue/babel-preset-jsx:  1.1.2 
    @vue/babel-sugar-functional-vue:  1.1.2 
    @vue/babel-sugar-inject-h:  1.1.2 
    @vue/babel-sugar-v-model:  1.1.2 
    @vue/babel-sugar-v-on:  1.1.2 
    @vue/cli-overlay:  4.2.2 
    @vue/cli-plugin-babel: ~4.2.0 => 4.2.2 
    @vue/cli-plugin-eslint: ~4.2.0 => 4.2.2 
    @vue/cli-plugin-router:  4.2.2 
    @vue/cli-plugin-vuex:  4.2.2 
    @vue/cli-service: ~4.2.0 => 4.2.2 
    @vue/cli-shared-utils:  4.2.2 
    @vue/component-compiler-utils:  3.1.1 
    @vue/preload-webpack-plugin:  1.1.1 
    @vue/web-component-wrapper:  1.2.0 
    eslint-plugin-vue: ^6.1.2 => 6.2.1 
    vue: ^2.6.11 => 2.6.11 
    vue-eslint-parser:  7.0.0 
    vue-hot-reload-api:  2.3.4
    vue-loader:  15.9.0
    vue-style-loader:  4.1.2
    vue-template-compiler: ^2.6.11 => 2.6.11
    vue-template-es2015-compiler:  1.9.1
  npmGlobalPackages:
    @vue/cli: Not Found
```

### Steps to reproduce
`vue create hello-world`

`vue-cli-service build --target lib --inline-vue --name vueTrp`

### What is expected?
vue is inlined inside the lib

### What is actually happening?
vue is not inlined in the lib

---
The Result locks so:

```html
<meta charset="utf-8">
<title>vueTrp demo</title>
<script src="https://unpkg.com/vue"></script>
<script src="./vueTrp.umd.min.js"></script>

<div id="app">
  <demo></demo>
</div>

<script>
new Vue({
  components: {
    demo: vueTrp
  }
}).$mount('#app')
</script>
```

But I don't want to load Vue again. If i comment the line out 

`<script src="https://unpkg.com/vue"></script>`

I get this error:
`demo:12 Uncaught ReferenceError: Vue is not defined`

WHY?

**Comment by luomavaltteri** [comment]
I'm wondering the same thing. Tried also to change the entry file to main.ts (like instructed here https://cli.vuejs.org/guide/build-targets.html#vue-vs-js-ts-entry-files), but wasn't able to get the application to render. I added the mentioned webpack config & created empty app, with following main.ts: 
```
...
const app = new Vue({
  store,
  render: h => h(App)
}).$mount("#app");

export default app;
```
With command `vue-cli-service build --target lib --name myLib --inline-vue src/main.ts`, shouldn't this render the application, if I have following HTML: 
```
<script src="./myLib.umd.js"></script>
<link rel="stylesheet" href="./myLib.css">
<div id="app"></div>
<script type="module">
  myLib.default;
</script>
```

Am I missing something?

**Comment by jaireina** [comment]
Hi @dertuerke,

Passing --inline-vue does include Vue as part of your generated lib (This is achieved by the [Externals](https://webpack.js.org/configuration/externals/) configuration option from Webpack). However, it does not makes the Vue variable available in the main window object since that would defeat the whole purpose of modularization. That's why you get the 'undefined' error when trying to use the Vue constructor in a page that doesn't include Vue. 

And @luomavaltteri, in your case you're including your library before the element with id app is loaded on the page, so that's why you never see your app rendering. Try placing the script tag that loads your library after the div#app and it should work.

**Comment by luomavaltteri** [comment]
Thanks @jaireina, that was it 😅

**Comment by johnnadratowski** [comment]
I stumbled upon this issue as I'm having similar problems inlining vue.  I am building the bundle with `--inline-vue` in UMD mode. However, in the build output, it's still trying to initialize itself referencing a global Vue reference on window.  Here's the code from the UMD bundle:

```
(function webpackUniversalModuleDefinition(root, factory) {
	if(typeof exports === 'object' && typeof module === 'object')
		module.exports = factory(require("vue"));
	else if(typeof define === 'function' && define.amd)
		define([], factory);
	else if(typeof exports === 'object')
		exports["app"] = factory(require("vue"));
	else
		root["app"] = factory(root["Vue"]);
})((typeof self !== 'undefined' ? self : this), function(__WEBPACK_EXTERNAL_MODULE__8bbf__) {
```

In this case, `root` is window and on this line: 

`root["app"] = factory(root["Vue"]);`

I'm building it with this command:

`yarn build --target lib --inline-vue --name app ./src/init.ts`

It's trying to reference a global `Vue` variable.  I'm so confused, please help!

Thanks.

**Comment by johnnadratowski** [comment]
You know, I just looked at the output and regardless whether or not I specify `--inline-vue`, the bundle output size is the same.  Strange as I'm definitely importing `Vue` from `init.ts`

**Comment by jaireina** [comment]
@johnnadratowski can you provide a repo where I can reproduce this?

**Comment by johnnadratowski** [comment]
Unfortunately I cannot.  I did create this repo based on another repo, which was built ~6 months ago. It could be that, during the process of copying + cleaning the code, then upgrading the packages, this broke.

This isn't super critical for me, I was doing some research to see if I could embed our vue app into a legacy UI app.  This was actually successful, as long as I included Vue from a CDN in my tests in the <head>.

Thanks!
